src/e2e-test/java/io/cdap/plugin/sfmcsource/actions/SfmcSourcePropertiesPageActions.java

```diff
@@ -16,32 +16,176 @@
+    Type type = new TypeToken<Map<String, Object>>() {
+    }.getType();
+    Map<String, Object> sfmcResponseInmap = gson.fromJson(sfmcResponse, type);
+    Map<String, Object> bigQueryResponseInMap = gson.fromJson(bigQueryResponse, type);
+    MapDifference<String, Object> mapDifference = Maps.difference(sfmcResponseInmap, bigQueryResponseInMap);
+    logger.info("Assertion :" + mapDifference);
+
+    return mapDifference.areEqual();
+  }
+
+  private static void getBigQueryTableData(String dataset, String table, String uniqueId)
+    throws IOException, InterruptedException {
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t WHERE " +
+      "id='" + uniqueId + "' ";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryrows.add(value.get(0).getValue().toString()));
+  }
+
+  private static TableResult getTableNamesFromDataSet(String bqTargetDataset) throws IOException, InterruptedException {
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + bqTargetDataset +
+      "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static void verifyIfRecordCreatedInSinkForMultipleObjectsAreCorrect(String expectedOutputFile)
+    throws IOException, InterruptedException {
+    List<String> expectedOutput = new ArrayList<>();
+    try (BufferedReader bf1 = Files.newBufferedReader(Paths.get(PluginPropertyUtils.pluginProp(expectedOutputFile)))) {
+      String line;
+      while ((line = bf1.readLine()) != null) {
+        expectedOutput.add(line);
+      }
+    }
+
+    List<String> bigQueryDatasetTables = new ArrayList<>();
+    TableResult tablesSchema = getTableNamesFromDataSet(dataset);
+    tablesSchema.iterateAll().forEach(value -> bigQueryDatasetTables.add(value.get(0).getValue().toString()));
+
+    for (int expectedRow = 0; expectedRow < expectedOutput.size(); expectedRow++) {
+      JsonObject expectedOutputAsJson = gson.fromJson(expectedOutput.get(expectedRow), JsonObject.class);
+      String uniqueId = expectedOutputAsJson.get("id").getAsString();
+      getBigQueryTableData(dataset, bigQueryDatasetTables.get(0), uniqueId);
+    }
+    for (int row = 0; row < bigQueryrows.size() && row < expectedOutput.size(); row++) {
+      Assert.assertTrue(compareValueOfBothResponses(expectedOutput.get(row), bigQueryrows.get(row)));
+    }
+  }
+
+  private static TableResult getDataExtensionUniqueIdFromBigQuery(String dataset, String table)
+    throws IOException, InterruptedException {
+    String id = TestSetupHooks.storeid;
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t WHERE " +
+      "storeid='" + id + "' ";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+
+    return result;
+  }
+
+  public static void verifyIfRecordsCreatedInSfmcSinkAreCorrect() throws ETSdkException,
```

Comment:
sfmc sink actions don't belong here

Reply:
Created a new Package for Sink and shifted the Steps and Actions there. (6f0d5e764e757b7af256ed710be646fbe221db6f)